src/runs.rs

```diff
@@ -0,0 +1,341 @@
+
+            // Create and save the infos file
+            let infos = DaemonRunInfos {
+                name: self.name.clone(),
+                task,
+                start_date: Local::now(),
+            };
+            let infos_json = serde_json::to_string_pretty(&infos).unwrap();
+            std::fs::write(self.infos_file_path(), infos_json).unwrap();
+
+            // Create the daemon
+            let daemonize = Daemonize::new()
+                .pid_file(self.pid_file_path())
+                .stdout(stdout)
+                .stderr(stderr)
+                .umask(0o027) // Set umask, `0o027` by default.
+                .chown_pid_file(true)
+                .working_directory(self.working_dir.clone());
+
+            // Start the daemon
+            daemonize.start().into_diagnostic()
+        }
+        #[cfg(not(unix))]
+        {
+            let _ = task;
+            miette::bail!("The `start` command is only available on Unix systems.");
+        }
+    }
+
+    pub fn clear(&self) -> miette::Result<()> {
+        // check if the run is running
+        if self.is_running() {
+            miette::bail!("The run is still running. You can call `pixi runs kill` to kill it.");
+        }
+
+        // delete pid, infos, stdout and stderr files
+        std::fs::remove_file(self.pid_file_path()).expect("Failed to remove pid file");
+        std::fs::remove_file(self.infos_file_path()).expect("Failed to remove infos file");
+        std::fs::remove_file(self.stdout_path()).expect("Failed to remove stdout file");
+        std::fs::remove_file(self.stderr_path()).expect("Failed to remove stderr file");
+
+        Ok(())
+    }
+
+    pub fn state(&self) -> miette::Result<DaemonRunState> {
+        let pid = match self.read_pid() {
+            Some(pid) => pid,
+            None => miette::bail!("No pid file with name '{}' found.", self.name),
+        };
+        let stdout_length = std::fs::read_to_string(self.stdout_path()).unwrap().len();
+        let stderr_length = std::fs::read_to_string(self.stderr_path()).unwrap().len();
+        let infos = match self.read_infos() {
+            Some(infos) => infos,
+            None => miette::bail!("No infos file with name '{}' found.", self.name),
+        };
+
+        Ok(DaemonRunState {
+            name: infos.name,
+            status: self.process_status(),
+            pid: pid.as_u32(),
+            task: infos.task,
+            start_date: infos.start_date,
+            stdout_length,
+            stderr_length,
+        })
+    }
+
+    pub fn kill(&self) -> miette::Result<()> {
+        let pid = match self.read_pid() {
+            Some(pid) => pid,
+            None => miette::bail!("No pid file with name '{}' found.", self.name),
+        };
+
+        // TODO: not very efficient to call this every time
+        let mut system = System::new_all();
+        system.refresh_all();
+
+        match system.process(pid) {
+            Some(process) => match process.kill() {
```

Comment:
When calling that cli on the jupyterlab example, the parent pixi process get killed but the child `jupyter lab` is still alive (I can continue using it).

Reply:
Indeed, `jupyterlab` also keeps running for me.

Reply:
Even the example in `examples/daemon/` does not get killed. I will investigate (and try `exec` as well).

Reply:
Should be working now. See comment below.